Panic (index out of range [-1]) in CalcCellValue when formula contains an invalid row reference (e.g., ROW($0:$9))

### Description

When attempting to calculate the value of a cell that contains a formula with an invalid row reference (specifically, referencing row 0 such as ROW($0:$9)), the CalcCellValue function triggers an unhandled panic: runtime error: index out of range [-1].
This occurs because Excel row indices are 1-based, and when excelize's rangeResolver attempts to parse and offset $0, it results in a negative index (-1) without bounds checking. This unhandled panic will abruptly crash the entire host Go application.

### Steps to reproduce the issue

Here is a minimal, reproducible Go script using v2.10.0:

```go
package main

import (
	"fmt"
	"github.com/xuri/excelize/v2"
)

func main() {
	f := excelize.NewFile()
	
	// Set a formula that contains an invalid row reference ($0)
	err := f.SetCellFormula("Sheet1", "C2", `=IFERROR(SMALL(IF(ISNUMBER(FIND(ROW($0:$9),TEXT(A2,"0"))),ROW($0:$9)),1),"")`)
	if err != nil {
		fmt.Println("Error setting formula:", err)
		return
	}
	
	fmt.Println("Triggering CalcCellValue...")
	
	// This call will trigger the panic: runtime error: index out of range [-1]
	_, err = f.CalcCellValue("Sheet1", "C2")
	
	fmt.Println("Done or error:", err)
}
```

The process crashes with the following panic trace:
```go

panic: runtime error: index out of range [-1]goroutine 1 [running]:github.com/xuri/excelize/v2.(*File).rangeResolver(...)    .../pkg/mod/github.com/xuri/excelize/v2@v2.10.0/calc.go:1727 +0x8d7github.com/xuri/excelize/v2.(*File).parseReference(...)    .../pkg/mod/github.com/xuri/excelize/v2@v2.10.0/calc.go:1588 +0xa73github.com/xuri/excelize/v2.(*File).evalInfixExp(...)    .../pkg/mod/github.com/xuri/excelize/v2@v2.10.0/calc.go:1020 +0xcc8github.com/xuri/excelize/v2.(*File).calcCellValue(...)    .../pkg/mod/github.com/xuri/excelize/v2@v2.10.0/calc.go:886 +0x171github.com/xuri/excelize/v2.(*File).CalcCellValue(...)    .../pkg/mod/github.com/xuri/excelize/v2@v2.10.0/calc.go:847 +0x334
```

## Environment Details

Go Version: go1.23+ (or your current version)
Excelize Version: v2.10.0
OS: macOS / Linux / Windows


### Describe the results you received

panic: runtime error: index out of range [-1]

goroutine 1 [running]:
github.com/xuri/excelize/v2.(*File).rangeResolver(...)
	.../pkg/mod/github.com/xuri/excelize/v2@v2.10.0/calc.go:1727 +0x8d7
github.com/xuri/excelize/v2.(*File).parseReference(...)
	.../pkg/mod/github.com/xuri/excelize/v2@v2.10.0/calc.go:1588 +0xa73
github.com/xuri/excelize/v2.(*File).evalInfixExp(...)
	.../pkg/mod/github.com/xuri/excelize/v2@v2.10.0/calc.go:1020 +0xcc8
github.com/xuri/excelize/v2.(*File).calcCellValue(...)
	.../pkg/mod/github.com/xuri/excelize/v2@v2.10.0/calc.go:886 +0x171
github.com/xuri/excelize/v2.(*File).CalcCellValue(...)
	.../pkg/mod/github.com/xuri/excelize/v2@v2.10.0/calc.go:847 +0x334

### Describe the results you expected

The CalcCellValue function should gracefully handle the invalid reference and return a standard Go error (e.g., #REF! or a parsing error), rather than panicking and bringing down the host process.

### Go version

go1.23+

### Excelize version or commit ID

v2.10.0

### Environment

```shell
OS: macOS / Microsoft Excel
```

### Validations

- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [x] The provided reproduction is a minimal reproducible example of the bug.

--- a/calc.go
+++ b/calc.go
@@ -1542,7 +1542,7 @@ func parseRef(ref string) (cellRef, bool, bool, error) {
 		}
 		if cr.Row, rowErr = strconv.Atoi(cell); rowErr == nil { // cast to row
 			if cr.Row < 1 || cr.Row > TotalRows {
-				return cr, false, false, fmt.Errorf("invalid row number %q", cell)
+				return cr, false, false, err
 			}
 			return cr, false, true, nil
 		}
@@ -1780,7 +1780,7 @@ func (f *File) rangeResolver(ctx *calcContext, cellRefs, cellRanges *list.List) 
 
 		for row := valueRange[0]; row <= valueRange[1]; row++ {
 			colMax := 0
-			if row >= 1 && row <= len(ws.SheetData.Row) {
+			if 0 < row && row <= len(ws.SheetData.Row) {
 				rowData := &ws.SheetData.Row[row-1]
 				colMax = min(valueRange[3], len(rowData.C))
 			}

--- a/calc_test.go
+++ b/calc_test.go
@@ -2908,6 +2908,8 @@ func TestCalcCellValue(t *testing.T) {
 		"SUM(1/)":           {ErrInvalidFormula.Error(), ErrInvalidFormula.Error()},
 		"SUM(1*SUM(1/0))":   {"#DIV/0!", "#DIV/0!"},
 		"SUM(1*SUM(1/0)*1)": {"", "#DIV/0!"},
+		"SUM(0:2)":          {"#NAME?", "invalid reference"},
+		"SUM(1:1048577)":    {"#NAME?", "invalid reference"},
 		// SUMIF
 		"SUMIF()": {"#VALUE!", "SUMIF requires at least 2 arguments"},
 		// SUMSQ

